plugins/generator-1.16.4/forge-1.16.4/procedures/entity_spawn_x.java.ftl

```diff
@@ -1,14 +1,14 @@
 (new Object() {
 	public int getSpawnX(World _world, Entity _ent) {
-		int retval = _world.getSpawnPoint().getX();
+		int retval = _world.getWorldInfo().getSpawnX();
 		if (_ent instanceof PlayerEntity) {
-			BlockPos _bp = ((PlayerEntity) _ent).getBedLocation(_world.getDimension().getType());
+			Optional<BlockPos> _bp = ((PlayerEntity) _ent).getBedPosition();
 			if (_bp != null)
```

Comment:
Optionals can't be null, but they can be empty. Please do proper optional null check here.

Reply:
I modified the check!